### PR TITLE
Animation for calendar when event created

### PR DESCRIPTION
**Pull request by Pitcity**

### CODE REVIEW

File: src/main/java/ua/softserveinc/tc/validator/EventValidator.java [hunk 1]
```diff
@@ -58,6 +60,15 @@ public boolean isReccurrentValid(RecurrentEventDto recurrentEventDto) {
         }
     }
 
+    public boolean isMonthlyValid(MonthlyEventDto monthlyEventDto) {
+        try {
+            dateSet(monthlyEventDto);
+            return onlyDateSimpleDateFormat.parse(monthlyEventDto.getStartTime()).before(onlyDateSimpleDateFormat.parse(monthlyEventDto.getEndTime()));
+        } catch (ParseException e) {
+            return false;
```
Comment (andryspo) [inline src/main/java/ua/softserveinc/tc/validator/EventValidator.java]: log exception

File: src/main/webapp/resources/js/renderCalendar.js [hunk 2]
```diff
@@ -634,9 +609,55 @@ function popSetOfEvents(set) {
...
+    startTime = new Date(startTime);
+    endTime = new Date(endTime);
+    var findTime = '.fc-content .fc-time';
+    var findName = '.fc-content .fc-title';
+
+    function getTime(date) {
+        time = '';
+        if (date.getHours() < timeZone) {
+            time += (date.getHours() - timeZone + hoursInDay);
+        } else if ((date.getHours() - timeZone) < 10) {
```
Comment (andryspo) [inline src/main/webapp/resources/js/renderCalendar.js]: what does 10 mean? use constant

File: src/main/java/ua/softserveinc/tc/controller/util/ViewEventController.java [hunk 3]
```diff
@@ -148,9 +150,13 @@ public String gerRecurrent(@RequestBody RecurrentEventDto recurrentEventDto, Bin
     }
 
     @PostMapping(value = "getmonthlyevents", produces = "application/json")
-    @ResponseBody
```
Comment (hrybs) [inline src/main/java/ua/softserveinc/tc/controller/util/ViewEventController.java]: You have deleted @ResponseBody. How can you send ResponseEntity?

File: src/main/java/ua/softserveinc/tc/service/impl/CalendarServiceImpl.java [hunk 4]
```diff
@@ -133,15 +137,13 @@ public void add(Event event) {
...
-                /// / TODO: 21.12.2016 throwsException??
-
-                res.add(genericMapper.toDto(newRecurrentEvent));
+                res.add(newRecurrentEvent);
             }
             calendar.add(Calendar.WEEK_OF_YEAR, 1);
             calendar.set(Calendar.DAY_OF_WEEK, daysOFWeek.get("Mon"));
         }
-        return res;
+        eventDao.saveSetOfEvents(res);
```
Comment (hrybs) [inline src/main/java/ua/softserveinc/tc/service/impl/CalendarServiceImpl.java]: It is good that you improve that as I and Vitaliy told yesterday. Good implementation our thoughts.

File: src/main/java/ua/softserveinc/tc/controller/util/ViewEventController.java (hunk 3 above)
Comment (Pitcity) [inline src/main/java/ua/softserveinc/tc/controller/util/ViewEventController.java]: as i know responseBody makes ResponseEntity from the Json obj, so if i send ResponseEntity directly i don't need that annotation. I may be mistaken..

Comment (hrybs) [inline src/main/java/ua/softserveinc/tc/controller/util/ViewEventController.java]: You may be right. Please test it before merging.

**Pull request merged**